app/client/components/productDetailWrapper.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from "react";
+import {
+  isProduct,
+  MembersDataApiItem,
+  MembersDatApiAsyncLoader,
+  ProductDetail
+} from "../../shared/productResponse";
+import { createProductDetailFetcher } from "../../shared/productTypes";
+import {
+  RouteableStepProps,
+  visuallyNavigateToParent
+} from "./wizardRouterAdapter";
+
+interface ProductDetailWrapperProps extends RouteableStepProps {
+  children: (productDetail: ProductDetail) => JSX.Element;
+  allowCancelledSubscription?: true;
+  loadingMessagePrefix: string;
+}
+
+export const ProductDetailWrapper = (props: ProductDetailWrapperProps) => {
```

Comment:
i'm having second thoughts about this name, as it's not wrapping ProductDetail it's wrapping flows, so it should it go back to...
```suggestion
export const ProductDetailProvider = (props: ProductDetailProviderProps) => {
```
... as this represents its current purpose/behaviour (despite the slight naming clash with Contexts).

**Then if/when we refactor the flows to have this component add the page furniture (nav & header) then we can rename it to just `FlowWrapper`.**